### PR TITLE
Fix/profiler tests

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

- add `pytest.mark.profiler` to flaky `test_profiling_counts_not_equal`
- explicitly check content, not length
- small refactors: constants, helpers

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`aea hash all` and `aea packages lock --check`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: tests/test_helpers/test_profiling.py [hunk 1]
```diff
@@ -183,40 +170,26 @@ def output_function(report):
...
-
+        assert count_dict["created"] == expected_created
+        assert count_dict["present"] == expected_present
     finally:
         p.stop()
-        p.wait_completed(sync=True, timeout=20)
+        p.wait_completed(sync=True, timeout=TIMEOUT)
 
 
+@pytest.mark.profiling
```
Comment (Karrenbelt) [inline tests/test_helpers/test_profiling.py:192]: took awhile to notice, but the marker was missing

**Comment by Karrenbelt** [comment]
failures seen:
- https://github.com/valory-xyz/open-aea/actions/runs/3147766749/jobs/5117646610
  ```
  FAILED tests/test_mail/test_base.py::test_outbox_put
  ```
  -> ticket open on this #349 


- https://github.com/valory-xyz/open-aea/actions/runs/3147766749/jobs/5117646337
  ```
  FAILED tests/test_manager/test_manager.py::TestMultiAgentManagerMultiprocessMode::test_remove_running_agent
  --> aea.manager.manager.AgentRunProcessTask
  
      def stop(self) -> None:
          """Stop the task."""
          if not self.process:
              raise ValueError("Task not started!")  # pragma: nocover
      
          self._stop_event.set()
          self.process.join(self.PROCESS_JOIN_TIMEOUT)
          if self.is_running:  # pragma: nocover
              self.process.terminate()
              self.process.join(self.PROCESS_JOIN_TIMEOUT)
  >           raise ValueError(
                  f"process was not stopped within timeout: {self.PROCESS_JOIN_TIMEOUT} and was terminated"
              )
  E           ValueError: process was not stopped within timeout: 30 and was terminated
  ```
  -> an attempt was made to address this by increasing the timeout from 20 -> 30 in #340

File: tests/test_helpers/test_profiling.py [hunk 2]
```diff
@@ -93,21 +109,17 @@ def output_function(report):
...
-    m = Message()
     try:
-        wait_for_condition(lambda: result, timeout=20)
-
+        wait_for_condition(lambda: result, timeout=TIMEOUT)
         assert "Profiling details" in result
     finally:
         p.stop()
-        p.wait_completed(sync=True, timeout=20)
-    del m
```
Comment (dvilelaf) [inline tests/test_helpers/test_profiling.py:110]: We don't instantiate a Message now. Shouldnt we be calling create_dummies?

Comment (Karrenbelt) [reply tests/test_helpers/test_profiling.py:110]: this was not used in the assertion

**Review by dvilelaf — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**